middleware/proxy/upstream.go

```diff
@@ -209,7 +212,70 @@ func (u *staticUpstream) HealthCheckWorker(stop chan struct{}) {
 	}
 }
 
+func (upstream *staticUpstream) ProviderWorker(watcher provider.Watcher, stop <-chan struct{}) {
+	type msg struct {
+		msg provider.WatcherMsg
+		err error
+	}
+	resp := make(chan msg)
+	go func() {
+	outer:
+		for {
+			// blocks until there's a message from Watcher
+			m, err := watcher.Next()
+			select {
+			case resp <- msg{m, err}:
+			case <-stop:
+				close(resp)
+				break outer
+			}
+		}
+	}()
+
+	interval := 5
+	delay := func() {
+		log.Printf("Delaying %d seconds before next attempt.", interval)
```

Comment:
Log message does not contain enough information. In logs it might be unclear what this "Delaying" is about..